pkg/synthetictests/image_pulls.go

```diff
@@ -0,0 +1,76 @@
+package synthetictests
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+type eventRecognizerFunc func(event monitorapi.EventInterval) bool
+
+func matchEventForRegexOrDie(regex string) eventRecognizerFunc {
+	regExp := regexp.MustCompile(regex)
+	return func(e monitorapi.EventInterval) bool {
+		return regExp.MatchString(e.Message)
+	}
+}
+
+type singleEventCheckRegex struct {
+	testName      string
+	recognizer    eventRecognizerFunc
+	failThreshold int
+}
+
+// test goes through the events, looks for a match using the s.recognizer function,
+// if a match is found, and occurs more than s.failThreshold times, we mark it as a
+// flake (this allows us to see how often this symptom is happening).
+//
+func (s *singleEventCheckRegex) test(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	success := &junitapi.JUnitTestCase{Name: s.testName}
+	for _, e := range events {
+		if s.recognizer(e) {
+
+			// We found an instance of "Back-off pulling image"; if it happens more than some
+			// threshold, report it as a flake.
+			//
+			msg := fmt.Sprintf("%s - %s", e.Locator, e.Message)
+			eventDisplayMessage, times := getTimesAnEventHappened(msg)
+			if times > s.failThreshold {
+				output := fmt.Sprintf("event [%s] happened %d times\n", eventDisplayMessage, times)
+				failure := &junitapi.JUnitTestCase{
+					Name:      s.testName,
+					SystemOut: output,
+					FailureOutput: &junitapi.FailureOutput{
+						Output: output,
+					},
+				}
+				// Include a fake success so this will always be a "flake" for now.
+				return []*junitapi.JUnitTestCase{failure, success}
```

Comment:
fail and flake are different.  If we're above the failureThreshold, it should be a straight failure, no success and return.  You can also add a flakeThreshold member and if we're above the flakeThreshold, we do this logic here.